### PR TITLE
boards/samd21-xpro: provide initial support

### PR DESCRIPTION
**Pull request by travisgriggs**

### CODE REVIEW

File: boards/samd21-xpro/Makefile.dep [hunk 1]
```diff
@@ -0,0 +1,7 @@
+ifneq (,$(filter gnrc_netdev_default netdev_default,$(USEMODULE)))
+  USEMODULE += at86rf233
+endif
```
Comment (thomaseichinger) [inline boards/samd21-xpro/Makefile.dep]: As far as I see this board does not have radio transceiver, this should be omitted then.

File: boards/samd21-xpro/include/board.h [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+/** @} */
+
+/**
+ * @brief   LED pin definitions and handlers
+ * @{
+ */
+#define LED0_PIN            GPIO_PIN(PB, 30)
+
+#define LED_PORT            PORT->Group[PB]
+#define LED0_MASK           (PB << 30)
```
Comment (thomaseichinger) [inline boards/samd21-xpro/include/board.h]: s/(PB << 30)/(1 << 30)/

File: boards/samd21-xpro/include/periph_conf.h [hunk 3]
```diff
@@ -0,0 +1,277 @@
...
+#define TIMER_1_MAX_VALUE   (0xffffffff)
+#define TIMER_1_ISR         isr_tc4
+/** @} */
+
+/**
+ * @name UART configuration
+ * @{
+ */
+static const uart_conf_t uart_config[] = {
+    {    // Virtual COM Port
```
Comment (thomaseichinger) [inline boards/samd21-xpro/include/periph_conf.h]: Here and following please use `/* C-style comments */`.

File: boards/samd21-xpro/include/periph_conf.h [hunk 4]
```diff
@@ -0,0 +1,277 @@
...
+
+#define I2C_0_DEV           SERCOM2->I2CM
+#define I2C_0_IRQ           SERCOM2_IRQn
+#define I2C_0_ISR           isr_sercom2
+/* I2C 0 GCLK */
+#define I2C_0_GCLK_ID       SERCOM2_GCLK_ID_CORE
+#define I2C_0_GCLK_ID_SLOW  SERCOM2_GCLK_ID_SLOW
+/* I2C 0 pin configuration */
+#define I2C_0_SDA           GPIO_PIN(PA, 08)
+#define I2C_0_SCL           GPIO_PIN(PA, 09)
```
Comment (thomaseichinger) [inline boards/samd21-xpro/include/periph_conf.h]: I get a compilation error here complaining about the leading 0 on arm-none-eabi-gcc 5.4.1.

**Comment by travisgriggs** [comment]
I resolved a conflict. And cleaned up some newbie errors I made in commit workflow. I'm confused by the CI static-tests error though. Do I need to squash?

File: boards/samd21-xpro/include/periph_conf.h [hunk 5]
```diff
@@ -0,0 +1,277 @@
...
+#define UART_2_ISR          isr_sercom5
+
+#define UART_NUMOF          (sizeof(uart_config) / sizeof(uart_config[0]))
+/** @} */
+
+/**
+ * @name PWM configuration
+ * @{
+ */
+
```
Comment (aabadie) [inline boards/samd21-xpro/include/periph_conf.h]: empty line, please remove

File: boards/samd21-xpro/board.c [hunk 6]
```diff
@@ -0,0 +1,37 @@
+/*
+ * Copyright (C) 2017 Travis Griggs
+ * Copyright (C) 2017 Dan Evans
```
Comment (lebrush) [inline boards/samd21-xpro/board.c]: Please see existing files for the format of copyright notes.

File: boards/samd21-xpro/include/gpio_params.h [hunk 7]
```diff
@@ -0,0 +1,55 @@
...
+
+/**
+ * @ingroup   boards_samd21-xpro
+ * @{
+ *
+ * @file
+ * @brief     Board specific configuration of direct mapped GPIOs
+ *
+ * @author    2017 Travis Griggs <travisgriggs@gmail.com>
+ * @author    2017 Dan Evans <photonthunder@gmail.com>
```
Comment (lebrush) [inline boards/samd21-xpro/include/gpio_params.h]: 2017?

File: boards/samd21-xpro/include/periph_conf.h [hunk 8]
```diff
@@ -0,0 +1,277 @@
...
+#include "periph_cpu.h"
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+/**
+ * @brief   External oscillator and clock configuration
+ *
+ * For selection of the used CORECLOCK, we have implemented two choices:
```
Comment (lebrush) [inline boards/samd21-xpro/include/periph_conf.h]: Please change to something like: "there are two choices for selection of CORECLOCK"

File: boards/samd21-xpro/include/periph_conf.h (hunk 5 above)
Comment (travisgriggs) [inline boards/samd21-xpro/include/periph_conf.h]: Fixed, thanks!

File: boards/samd21-xpro/board.c (hunk 6 above)
Comment (travisgriggs) [inline boards/samd21-xpro/board.c]: Fixed, I think.

File: boards/samd21-xpro/include/gpio_params.h (hunk 7 above)
Comment (travisgriggs) [inline boards/samd21-xpro/include/gpio_params.h]: Copy/paste snafu. My bad. Fixed.

File: boards/samd21-xpro/include/periph_conf.h (hunk 8 above)
Comment (travisgriggs) [inline boards/samd21-xpro/include/periph_conf.h]: Reworded to remove the "we" as suggested. Also reworded another "we" verbiage further down.

**Comment by travisgriggs** [comment]
Sorry for the delay. Daughter's college graduation kind of interrupted things. With some helpful guidance, I've squashed down to the two recommended commits, and got all the changes requested, I hope.

File: boards/samd21-xpro/Makefile.include [hunk 9]
```diff
@@ -0,0 +1,25 @@
...
+    export OPENOCD_EXTRA_INIT += "-c cmsis_dap_serial $(SERIAL)"
+    SERIAL_TTY = $(firstword $(shell $(RIOTBASE)/dist/tools/usb-serial/find-tty.sh $(SERIAL)))
+    ifeq (,$(SERIAL_TTY))
+        $(error Did not find a device with serial $(SERIAL))
+    endif
+    PORT_LINUX := $(SERIAL_TTY)
+endif
+
+# this board uses openocd
+include $(RIOTBOARD)/Makefile.include.openocd
```
Comment (thomaseichinger) [inline boards/samd21-xpro/Makefile.include]: Please replace by `include $(RIOTMAKE)/tools/openocd.inc.mk`.

File: boards/samd21-xpro/Makefile.include [hunk 10]
```diff
@@ -0,0 +1,25 @@
+# define the cpu used by SAMD21 Xplained Pro board
+export CPU = samd21
+export CPU_MODEL = samd21j18a
+
+# set default port depending on operating system
+PORT_LINUX ?= /dev/ttyACM0
+PORT_DARWIN ?= $(firstword $(sort $(wildcard /dev/tty.usbmodem*)))
+
+# setup serial terminal
+include $(RIOTBOARD)/Makefile.include.serial
```
Comment (thomaseichinger) [inline boards/samd21-xpro/Makefile.include]: Please replace by `include $(RIOTMAKE)/tools/serial.inc.mk`.

**Comment by thomaseichinger** [comment]
@travisgriggs Thanks for updating and congratulations to your daugther! :)

PR #4699 moved and renamed some makefiles you include, could you please make changes accordingly?

**Comment by travisgriggs** [comment]
Sure thing @thomaseichinger. Is it still need to be collapsed into the two commits?

File: boards/samd21-xpro/include/board.h [hunk 11]
```diff
@@ -0,0 +1,76 @@
...
+#include "cpu.h"
+#include "periph_conf.h"
+#include "periph_cpu.h"
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+/**
+ * @brief   xtimer configuration
```
Comment (aabadie) [inline boards/samd21-xpro/include/board.h]: should be `@name` here and below (use `@name` with `@{ .. /** @} */` blocks, otherwise `@brief` is fine.

Comment (travisgriggs) [inline boards/samd21-xpro/include/board.h]: I've never authored any doxygen stuff myself @aabadie . I just basically cloned this from the other sam21 boards. I'm glad to learn though... is the basic rule here that if the block includes the @{ and an eventual closing @}, then it should be @name. And @brief can be used when there is no following block to be enclosed?

File: boards/samd21-xpro/include/periph_conf.h [hunk 12]
```diff
@@ -0,0 +1,275 @@
...
+
+#include "cpu.h"
+#include "periph_cpu.h"
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+/**
+ * @brief   External oscillator and clock configuration
```
Comment (aabadie) [inline boards/samd21-xpro/include/periph_conf.h]: `@name`

File: boards/samd21-xpro/include/board.h (hunk 11 above)
Comment (aabadie) [inline boards/samd21-xpro/include/board.h]: yes

Comment (travisgriggs) [inline boards/samd21-xpro/include/board.h]: Thanks. That helps.

**Comment by aabadie** [comment]
One last think before ACK: rebase and squash. @thomaseichinger anything remaining on you side ?

**Comment by travisgriggs** [comment]
@aabadie can you clarify how squashed? Do you want it in two commits as before? Or just one?

**Comment by aabadie** [comment]
Yes in 2 commits as before. Use `git rebase origin/master -i` (assuming origin is the up-to-date upstream remove, use `git fetch origin` if unsure) and play with the history of your git branch.

**Comment by travisgriggs** [comment]
Is there something I still need to fix here? The reported error is one that's showing in a bunch of PRs right now, and seems unrelated to the changes. Begin new to the process, is it just wait and hope now?

**Comment by thomaseichinger** [comment]
@travisgriggs Thanks for your fixes. I don't have the hardware so I can't test but I trust you for this. Needs rebase though because of `tests/unittests/Makefile`.

**Pull request merged**